engine/lib/analyses/mutable_variables.ml

```diff
@@ -0,0 +1,91 @@
+open Base
+open Utils
+
+module%inlined_contents Make (F : Features.T) = struct
+  module FA = F
+  module A = Ast.Make (F)
+  open Ast
+
+  type pre_data =
+    ( concrete_ident,
+      concrete_ident list,
+      Concrete_ident.comparator_witness )
+    Map.t
+    * (concrete_ident, int, Concrete_ident.comparator_witness) Map.t
```

Comment:
I think here a few things would help:
 1. Use `Map.M`, to get rid of the `comparator_witness` noise:
    ```suggestion
       type analysis_data =
           concrete_ident list Map.M(Concrete_ident).t * int Map.M(Concrete_ident).t
    ```
 2. Make a type synonym and use it instead of `int` (I guess it's some kind of id?)